script-generator/src/main/java/com/coolbitx/wallet/signing/scriptlib/SolScript.java

```diff
@@ -0,0 +1,196 @@
+                .showAddress(ScriptData.getDataBufferAll(Buffer.CACHE2))
+                .clearBuffer(Buffer.CACHE2)
+                .baseConvert( data, Buffer.CACHE1, 8, ScriptAssembler.binaryCharset, ScriptAssembler.inLittleEndian)
+                .showAmount(ScriptData.getDataBufferAll(Buffer.CACHE1), 9)
+                .clearBuffer(Buffer.CACHE1)
+                .showPressButton()
+                .setHeader(HashType.NONE, SignType.EDDSA)
+                .getScript();
+
+        return script;
+    }
+
+    public static String getTransferSelfScript() {
+        ScriptArgumentComposer sac = new ScriptArgumentComposer();
+        ScriptData fromAccount = sac.getArgument(32);
+        ScriptData programId = sac.getArgument(32);
+        ScriptData recentBlockHash = sac.getArgument(32);
+        ScriptData dataLength = sac.getArgument(1);
+        ScriptData programIdIndex = sac.getArgument(4);
+        ScriptData data = sac.getArgumentAll();
+
+        ScriptAssembler scriptAsb = new ScriptAssembler();
+        String script = scriptAsb
+                .setCoinType(0x01f5)
+                .copyString("01")
+                .copyString("00")
+                .copyString("01")
+                .copyString("02")
+                .copyArgument(fromAccount)
+                .copyArgument(programId)
+                .copyArgument(recentBlockHash)
+                .copyString("01")
+                .copyString("01")
+                .copyString("02")
+                .copyString("0000")
+                .copyArgument(dataLength)
+                .copyArgument(programIdIndex)
+                .copyArgument(data)
+                .showMessage("SOL")
+                .baseConvert( fromAccount, Buffer.CACHE2, 0, ScriptAssembler.base58Charset, ScriptAssembler.zeroInherit)
+                .showAddress(ScriptData.getDataBufferAll(Buffer.CACHE2))
+                .clearBuffer(Buffer.CACHE2)
+                .baseConvert( data, Buffer.CACHE1,8, ScriptAssembler.binaryCharset, ScriptAssembler.inLittleEndian)
+                .showAmount(ScriptData.getDataBufferAll(Buffer.CACHE1), 9)
+                .clearBuffer(Buffer.CACHE1)
+                .showPressButton()
+                .setHeader(HashType.NONE, SignType.EDDSA)
+                .getScript();
+        return script;
+    }
+
+    public static String getSolSmartScript() {
+        ScriptArgumentComposer sac = new ScriptArgumentComposer();
+        ScriptData keys = sac.getArgumentRightJustified(96);
+        ScriptData recentBlockHash = sac.getArgumentRightJustified(32);
+        ScriptData dataLength = sac.getArgument(1);
+        ScriptData data = sac.getArgumentAll();
+
+        ScriptAssembler scriptAsb = new ScriptAssembler();
+        String script = scriptAsb
+                .setCoinType(0x01f5)
+                .copyString("01")
+                .copyString("00")
+                .copyString("01")
+                .copyString("03")
```

Comment:
Key length of the smart contract will not always be `3`.

Reply:
Could you tell me more detail of which case smart contract key length wasn't be 3

Reply:
It depends on the contract.
ex. https://explorer.solana.com/address/M2mx93ekt1fmXSVkTrUL9xVFHkmME8HTUi5Cyc5aF7K